### PR TITLE
Decouple a bit user schemas from organizations

### PR DESCRIPTION
**Pull request by zenitraM**
We right now assume that the only way for an user to be on its own, non-public schema is when it's part of an organization.

However, there could be the case of a non-organization user being on its own schema. (most useful on the user/org migration process).

This decouples this behavior by:

- Moving some schema-migration methods from the organization controller to the user model
- Adding a move_user_to_schema method to the user
- Making database_publicuser no longer depend on if the user is an org or not to decide if it should use the common "publicuser" instead of the specific one for the user (Windshaft and SQL API do this properly)

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2392/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2395/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2397/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2400/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2401/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2414/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2492/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2493/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2536/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2549/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2561/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2568/))

File: app/models/user.rb [hunk 1]
```diff
@@ -483,7 +483,7 @@ def database_username
   end
 
   def database_public_username
-    has_organization_enabled? ? "cartodb_publicuser_#{id}" : CartoDB::PUBLIC_DB_USER
+    (self.database_schema != "public") ? "cartodb_publicuser_#{id}" : CartoDB::PUBLIC_DB_USER
```
Comment (Kartones) [inline app/models/user.rb]: Can you use a constant `PUBLIC_SCHEMA='public'`? is easier to search/grep than to do it for "public"

File: app/models/user.rb [hunk 2]
```diff
@@ -1498,13 +1498,33 @@ def setup_organization_user
...
     self.create_public_db_user
-    self.setup_schema
+  end
+
+  def move_to_own_schema
+    self.move_to_schema(self.username)
+  end
+
+  def move_to_schema(new_schema_name)
+    if self.database_schema != new_schema_name
```
Comment (Kartones) [inline app/models/user.rb]: Would probably be nice to make this raise an error if the schema already exists, just to fail as early as possible

**Comment by Kartones** [comment]
:+1: as you say there are other tests covering this code :smile:

File: app/models/user.rb (hunk 1 above)
Comment (zenitraM) [inline app/models/user.rb]: Fixed on commit coming in 5, 4, 3..

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2575/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2577/))

**Pull request merged**